From 031d2f1e8c06440966807fea26ff0553f3ea3de3 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Wed, 19 Nov 2014 13:08:30 +0000
Subject: [PATCH] MythUIHelper: Fix loading some themes after 5cd97d7fd9

We use the directory name to load themes not the name in the theme.xml!
---
 mythtv/libs/libmythui/mythuihelper.cpp |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythui/mythuihelper.cpp b/mythtv/libs/libmythui/mythuihelper.cpp
index 83a2a27..2e05db2 100644
--- a/mythtv/libs/libmythui/mythuihelper.cpp
+++ b/mythtv/libs/libmythui/mythuihelper.cpp
@@ -1152,12 +1152,14 @@ QStringList MythUIHelper::GetThemeSearchPath(void)
     bool found = true;
     QString themeName = d->m_themename;
     QString baseName;
+    QString dirName;
 
     while (found && !themeName.isEmpty())
     {
         // find the ThemeInfo for this theme
         found = false;
         baseName = "";
+        dirName = "";
 
         for (int x = 0; x < themeList.count(); x++)
         {
@@ -1165,6 +1167,7 @@ QStringList MythUIHelper::GetThemeSearchPath(void)
             {
                 found = true;
                 baseName = themeList.at(x).GetBaseTheme();
+                dirName = themeList.at(x).GetDirectoryName();
                 break;
             }
         }
@@ -1172,7 +1175,7 @@ QStringList MythUIHelper::GetThemeSearchPath(void)
         // try to find where the theme is installed
         if (found)
         {
-            QString themedir = FindThemeDir(themeName, false);
+            QString themedir = FindThemeDir(dirName, false);
             if (!themedir.isEmpty())
             {
                 LOG(VB_GUI, LOG_INFO, LOC + QString("Adding path '%1' to theme search paths").arg(themedir));
-- 
1.7.10.2

